client/verta/verta/_deployment/endpoint.py

```diff
@@ -171,21 +171,19 @@ def _update_from_build(self, build_id, strategy, wait=False, resources=None, aut
         if wait:
             print("waiting for update...", end='')
             sys.stdout.flush()
-            while self.get_status()['status'] not in ("active", "error"):
+
+            status_dict = self.get_status()  # if you think of a better var name, please do use it haha
+            while status_dict['status'] not in ("active", "error") \
+                    or (status_dict['status'] == "active" and len(self.get_status()['components']) > 1):
                 print(".", end='')
+                status_dict = self.get_status()
```

Comment:
Let's move this to after the `sleep` (so that the `get_status()` and the loop's condition aren't 5 seconds apart).

Reply:
Fixed. And god damn the test is slow :(